src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1394,21 +1391,30 @@ ucs_status_t uct_dc_mlx5_ep_check_fc(uct_dc_mlx5_iface_t *iface,
         fc_entry = &kh_value(&iface->tx.fc_hash, it);
 
         /* Do not resend FC request if timeout is not reached, or window is
-         * still not empty */
-        if (ucs_likely((ep->fc.fc_wnd > 0) ||
+         * still not empty (fc_wnd == fc_hard_thresh when FC_HARD_REQ packet
+         * was sent and the endpoint was added to FC hash, but the user data
+         * wasn't sent to a peer and fc_wnd wasn't decremented as a result) */
+        if (ucs_likely((ep->fc.fc_wnd ==
+                                iface->super.super.config.fc_hard_thresh) ||
```

Comment:
do we really need to check ep->fc.fc_wnd==fc_hard_thresh?
worse case we will send yet another hard_req after some time?

Reply:
> worse case we will send yet another hard_req after some time?

yes, after `iface->tx.fc_hard_req_timeout`
removed